lib/tic_tac_toe_checker.ex

```diff
@@ -6,8 +6,15 @@ defmodule TicTacToeChecker do
 
   use GenServer
 
+  @type row() :: [non_neg_integer()]
+  @type board() :: [row()]
+  @type state() :: %{board: board(), id: non_neg_integer(), results: list()}
```

Comment:
Feels like we need `@type result :: …` and here:
```suggestion
  @type state() :: %{board: board(), id: non_neg_integer(), results: [result()]}
```